### PR TITLE
Added configuration options to Microsoft.Batch/batchAccounts

### PR DESCRIPTION
**Pull request by fblix**
# Description

The current batchaccounts module does not support any detailed configurations towards the Azure resource.
The one @JPEasier and myself added, supports the full range of options currently available from the API directly.

Developed with pair-programming.

## Pipeline references
[Batch: BatchAccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.batch.batchaccounts.yml)

# Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

**Comment by fblix** [comment]
waiting for #1350

File: arm/Microsoft.Batch/batchAccounts/deploy.bicep [hunk 1]
```diff
@@ -32,6 +65,30 @@ param lock string = 'NotSpecified'
...
+  'TaskAuthenticationToken'
+])
+@description('Optional. List of allowed authentication modes for the Batch account that can be used to authenticate with the data plane.')
+param allowedAuthenticationModes array = []
+
+@allowed([
+  'Microsoft.Batch'
+  'Microsoft.KeyVault'
+])
+@description('Otional. Type of the key source.')
```
Comment (eriqua) [inline arm/Microsoft.Batch/batchAccounts/deploy.bicep:80]:
```suggestion
@description('Optional. Type of the key source.')
```
Please regenerate readme

File: arm/Microsoft.Batch/batchAccounts/.parameters/min.parameters.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "name": {
+            "value": "<<namePrefix>>azbaweux001"
```
Comment (eriqua) [inline arm/Microsoft.Batch/batchAccounts/.parameters/min.parameters.json:6]:
```suggestion
            "value": "<<namePrefix>>azbaweumin001"
```

File: .azuredevops/modulePipelines/ms.batch.batchaccounts.yml [hunk 3]
```diff
@@ -45,7 +45,7 @@ stages:
         parameters:
           removeDeployment: '${{ parameters.removeDeployment }}'
           deploymentBlocks:
-            - path: $(modulePath)/.parameters/parameters.json
+            - path: $(modulePath)/.parameters/min.parameters.json
```
Comment (eriqua) [inline .azuredevops/modulePipelines/ms.batch.batchaccounts.yml:48]:
```suggestion
            - path: $(modulePath)/.parameters/min.parameters.json
            - path: $(modulePath)/.parameters/parameters.json
```

File: arm/Microsoft.Batch/batchAccounts/readme.md [hunk 4]
```diff
@@ -36,7 +45,18 @@
...
 | `lock` | string | `'NotSpecified'` | `[CanNotDelete, NotSpecified, ReadOnly]` | Specify the type of lock. |
+| `poolAllocationMode` | string | `'BatchService'` | `[BatchService, UserSubscription]` | The allocation mode for creating pools in the Batch account. Determines which quota will be used. |
+| `publicNetworkAccess` | string | `'Enabled'` | `[Disabled, Enabled]` | The network access type for operating on the resources in the Batch account. |
+| `storageAccessIdentity` | string | `''` |  | The reference to a user assigned identity associated with the Batch pool which a compute node will use. |
+| `storageAuthenticationMode` | string | `'StorageKeys'` | `[BatchAccountManagedIdentity, StorageKeys]` | The authentication mode which the Batch service will use to manage the auto-storage account. |
+| `systemAssignedIdentity` | bool | `False` |  | Enables system assigned managed identity on the resource. |
 | `tags` | object | `{object}` |  | Tags of the resource. |
+| `userAssignedIdentities` | object | `{object}` |  | The ID(s) to assign to the resource. |
+
+**Otional parameters**
```
Comment (eriqua) [inline arm/Microsoft.Batch/batchAccounts/readme.md:56]: Please regenerate readme

Comment (fblix) [reply arm/Microsoft.Batch/batchAccounts/readme.md]: done

**Comment by JPEasier** [comment]
Waiting for Bug: #1347

**Comment by MrMCake** [comment]
> Waiting for Bug: #1347

As per the reply in the other PR - Linter should not block this PR as it is overly sensitive

**Comment by fblix** [comment]
> > Waiting for Bug: #1347
> 
> As per the reply in the other PR - Linter should not block this PR as it is overly sensitive

gotcha, thought the linter had to be green in order to merge it. Will proceed as soon as @eriqua gives her go :)

File: arm/Microsoft.Batch/batchAccounts/readme.md [hunk 5]
```diff
@@ -92,6 +97,7 @@ You can specify multiple user assigned identities to a resource by providing add
 | `resourceGroupName` | string | The resource group the batch account was deployed into |
 | `resourceId` | string | The resource ID of the batch account |
 
+
 ## Template references
```
Comment (eriqua) [inline arm/Microsoft.Batch/batchAccounts/readme.md:101]: I think with the latest changes to the readme generator, the template references section is not expected anymore. @MrMCake to confirm.

**Review by MrMCake — APPROVED** [review]

**Pull request merged**